src/main/java/org/example/services/implementations/UserServiceImpl.java

```diff
@@ -71,5 +74,24 @@ public void assignDepartmentToUser(int userId, String departmentName) {
         userRepository.save(user);
     }
 
+    @Transactional
+    @Override
+    public User assignRoleToUser(int userId, int roleId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("No such user by id " + userId));
+
+        Role role = roleRepository.findById(roleId)
+                .orElseThrow(() -> new RuntimeException("No such role by id " + roleId));
```

Comment:
Don't really matter, but it is better to have different Exception class for this, e.g. NotFoundException that extends the RuntimeExceptions. Later you could create a custom exception handler that will return different status codes depending on the exception thrown